docs/docs/reference/crd-reference/task.md

```diff
@@ -42,61 +44,63 @@ spec:
-      to the `KeptnWorkload` and `KeptnApp` resources.
-
-    * **taskType** (required) -- Indicates whether this `KeptnTask`
-      is part of the pre- or post-deployment phase.
-      When populating this resource manually
-      to run a task for a non-Kubernetes deployment,
-      set this value to `""`:
-      Keptn populates this field based on annotations
-      to the `KeptnWorkload` and `KeptnApp` resources.
-
-    * **workloadName** (required) -- Name of the `KeptnWorkload`
-      for which the `KeptnTask` is being executed.
-    * **workloadVersion** (required) -- Version of the `KeptnWorkload`
-      for which the `KeptnTask` is being executed.
-  * **parameters** -- Parameters that are passed to the job
-    that executes the `KeptnTask`.
-  * **secureParameters** -- Secure parameters that are passed
-    to the job that executes the `KeptnTask`.
-    These are stored and accessed as Kubernetes `Secrets` in the cluster.
-    See [Working with secrets](../../guides/tasks.md#working-with-secrets)
-    for more information.
-  * **checkType** -- Defines whether task is part of pre- or post-deployment phase.
-    Keptn populates this field based on annotations
-    to the `KeptnWorkload` and `KeptnApp` resources.
-  * **retries** -- If errors occur,
-    this defines the number of attempts made
-    before the `KeptnTask` is considered to be failed.
-  * **timeout** -- Specifies the time, in seconds,
-    to wait for the `KeptnTask` to complete successfully.
-    If the `KeptnTask` does not complete successfully in this timeframe,
-    it is considered to be failed.
+    * **name** -- Unique name of this run of the task.
+      This name must be modified each time you run this `KeptnTask`,
+      so a common practice is to add a number to the end of the string
+      so you can increment the number for each run.
+      Names must comply with the
+      [Kubernetes Object Names and IDs](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names)
+      specification.
+        * **spec** - Defines the specification of this `KeptnTask` resource
+            * **taskDefinition** (required) -- Name of the corresponding `KeptnTaskDefinition` resource.
+              This `KeptnTaskDefinition` can be located in the same namespace
+              or in the Keptn installation namespace.
+            * **context** (required) -- Contextual information about the task execution
+                * **appName** (required) -- Name of the
+                  [KeptnApp](app.md) resource
+                  for which the `KeptnTask` is being executed.
+                * **appVersion** (required) -- Version of the `KeptnApp` resource
+                  for which the `KeptnTask` is being executed.
+                * **metadata** -- Additional key-value pairs with contextual information for the `KeptnTask`.
+                  Keptn populates this field based on the metadata field of
```

Comment:
it should probably be `spec.metadata` since `metadata` is the k8s built-in field
```suggestion
                  Keptn populates this field based on the `spec.metadata` field of
```

Reply:
fair point!